common/src/types/v0/store/pool.rs

```diff
@@ -47,6 +47,17 @@ impl ResourceUuid for PoolState {
 /// Status of the Pool Spec
 pub type PoolSpecStatus = SpecStatus<message_bus::PoolStatus>;
 
+impl From<i32> for PoolSpecStatus {
+    fn from(i: i32) -> Self {
+        match i {
+            1 => PoolSpecStatus::Created(Default::default()),
+            2 => PoolSpecStatus::Deleting,
+            3 => PoolSpecStatus::Deleted,
+            _ => PoolSpecStatus::Creating,
```

Comment:
Should we be setting the catch all to Creating? If it is possible to have an "unsupported" value, should we handle it differently?

Reply:
To handle this, I have used the corresponding grpc type to match each arm, so that we dont have a catch all thing.